pkg/authorization/apiserver/admission/restrictusers/restrictusers.go

```diff
@@ -167,32 +167,30 @@ func (q *restrictUsersAdmission) Validate(a admission.Attributes, _ admission.Ob
 		return nil
 	}
 
-	if !q.waitForSyncedStore(time.After(timeToWaitForCacheSync)) {
-		return admission.NewForbidden(a, errors.New("rbr cache not synchronized"))
-	}
-
-	roleBindingRestrictionList, err := q.rbrLister.List(labels.Everything())
+	// TODO: Cache rolebinding restrictions.
+	roleBindingRestrictionList, err := q.roleBindingRestrictionsGetter.RoleBindingRestrictions(ns).
+		List(metav1.ListOptions{})
 	if err != nil {
-		return admission.NewForbidden(a, fmt.Errorf("could not list rolebinding restrictions: %v", err))
+		return admission.NewForbidden(a, err)
 	}
-	if len(roleBindingRestrictionList) == 0 {
+	if len(roleBindingRestrictionList.Items) == 0 {
 		klog.V(4).Infof("No rolebinding restrictions specified; admitting")
 		return nil
 	}
 
 	checkers := []SubjectChecker{}
-	for _, rbr := range roleBindingRestrictionList {
+	for _, rbr := range roleBindingRestrictionList.Items {
 		checker, err := NewSubjectChecker(&rbr.Spec)
 		if err != nil {
-			return admission.NewForbidden(a, fmt.Errorf("could not create rolebinding restriction subject checker: %v", err))
```

Comment:
We still want the better error messages here.

Reply:
got it